include/pybind11/numpy.h

```diff
@@ -500,6 +574,36 @@ class array : public buffer {
         return offset_at(index...) / itemsize();
     }
 
+    /** Returns a proxy object that provides access to the array's data without bounds or
+     * dimensionality checking.  Will throw if the array is missing the `writeable` flag.  Use with
+     * care: the array must not be destroyed or reshaped for the duration of the returned object,
+     * and the caller must take care not to access invalid dimensions or dimension indices.
+     */
+    template <typename T, size_t Dims> detail::unchecked_reference<T, Dims> unchecked() {
```

Comment:
Good point.  Additionally with the `unchecked()` -> `mutable_unchecked()` rename it makes no sense at all to have a `mutable_unchecked()` that returns a non-mutable reference.